src/core/mac/data_poll_handler.cpp

```diff
@@ -273,10 +273,10 @@ void DataPollHandler::HandleSentFrame(const Mac::TxFrame &aFrame, Error aError,
                 uint32_t frameCounter;
                 uint8_t  keyId;
 
-                IgnoreError(aFrame.GetFrameCounter(frameCounter));
+                OT_ASSERT(aFrame.GetFrameCounter(frameCounter) == kErrorNone);
                 aChild.SetIndirectFrameCounter(frameCounter);
 
-                IgnoreError(aFrame.GetKeyId(keyId));
+                OT_ASSERT(aFrame.GetKeyId(keyId) == kErrorNone);
```

Comment:
I wonder what problem was detected with the use of `IgnoreError()` here?

Reply:
Both this and GetFrameCounter can fail on FindSecurityHeaderIndex(); leading to output parameter not being initialized.
If the frame would be created somewhere around this calls I would verify if it is properly set and mark as False Positive in our system if so. However if I am not mistaken we are in the callback for transmission being finished and we should not assume it was not malformed on the way e.g while being handled by the radio driver.

Reply:
I reverted this change as VerifyOrExit around GetFrameCounter also covers the fail scenario for this call i.e. they will always both fail or both pass.

Reply:
Same as above.
```
error = aFrame.GetKeyId(keyId);
OT_ASSERT(error == kErrorNone);
```